city_scrapers/spiders/il_sex_offender_management.py

```diff
@@ -0,0 +1,234 @@
+                )
+
+    def _parse_meta_video(self, response):
+        """Parse through meta of response in JSON if link provided rather than PDF"""
+        meta_text = response.css(
+            "[type='application/json']#extendedData::text"
+        ).extract_first()
+        if not meta_text:
+            return
+        meta_text = meta_text.strip()
+        clean_text_vid = json.loads(meta_text)
+
+        start_time, end_time = self._video_start_end_time(clean_text_vid)
+
+        meeting = Meeting(
+            title="MANAGEMENT BOARD",
+            description="",
+            classification=BOARD,
+            start=start_time,
+            end=end_time,
+            all_day=False,
+            time_notes="",
+            location={
+                "address": "",
+                "name": "WebEx Meeting (See meeting links to confirm)",
+            },
+            links=self._make_links(None, start_time),
+            source=response.url,
+        )
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _video_start_end_time(self, clean_text_vid):
+        """Parse through JSON and return start, end naive datetime objects"""
+
+        # get date time formatted as '2020-09-16 13:00'
+        date_time = clean_text_vid["meetingData"]["startTime"][:-3]
+
+        # get meeting duration represented in minutes
+        scheduled_duration = clean_text_vid["meetingData"]["scheduledDuration"]
+
+        # create start and end time datetime naive objects
+        start_time = datetime.strptime("{}".format(date_time), "%Y-%m-%d %H:%M")
+
+        end_time = start_time + timedelta(minutes=scheduled_duration)
+
+        return start_time, end_time
+
+    def _parse_documents(self, response):
+        """Parse meeting information from agenda PDF"""
+
+        links_dict = {
+            "title": "Meeting Agenda",
+            "href": response.url,
+        }
+        clean_text = self._cleanUpPDF(response)
+
+        start_time = self._parse_start(clean_text)
+
+        meeting = Meeting(
+            title="MANAGEMENT BOARD",
+            description="",
+            classification=BOARD,
+            start=start_time,
+            end=self._parse_end(clean_text),
+            all_day=False,
+            time_notes="See agenda to confirm details",
+            location=self._parse_location(clean_text),
+            links=self._make_links(links_dict, start_time),
+            source=self.start_urls[0],
+        )
+        meeting["status"] = self._get_status(meeting, text=clean_text)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _make_links(self, agenda_dict, date_time_obj):
+        """Make list of agenda and minutes(if applicable)."""
+        links = []
+        if agenda_dict is not None:
+            links.append(agenda_dict)
+        for meeting in self.meeting_minutes:
```

Comment:
We've used `collections.defaultdict` for this in the past which has made it easier to check for links and return an empty list if they aren't present. We're doing that in [`chi_lsc_advisory`](https://github.com/City-Bureau/city-scrapers/blob/c42620480f9184ca74885e2a3c6ef576faba3eb7/city_scrapers/spiders/chi_lsc_advisory.py)